n_t_with_s/Snakefile

```diff
@@ -0,0 +1,361 @@
+
+"""
+Snakefile for running N_t analysis on stdpopsim.
+
+Simply running `snakemake` will run all analysis 
+defined by the arguments above.
+
+Currently, for each rep,
+This will run stairway plot, smc++, and msmc
+on the data resulting from simulations
+on all chromosomes included in chrm_list
+for estimates of N_t (Ne through time).
+"""
+
+import pathlib
+import sys
+import os
+
+import numpy as np
+import stdpopsim
+
+import stairway
+import smc
+import msmc
+import plots
+
+
+# ###############################################################################
+# KNOBS - 
+# ###############################################################################
+
+
+# A seed to replicate results
+# TODO mutation rates
+
+configfile: config["config"]+"/config.json"
+
+np.random.seed(config["seed"])
+
+# This is the number of samples to simulate for within each population
+# for each replicate
+# TODO double check this is up to date with stdpopsim backend 
+num_samples_per_population = config["num_samples_per_population"]
+population_id = config["population_id"]
+# Here is a list of sample sizes to run msmc on. 
+# Each element counts as its own analysis
+# so there will be "replicates" runs for each size
+num_sampled_genomes_msmc = [int(x) for x in config["num_sampled_genomes_msmc"].split(",")]
+
+# The number of msmc Baumwelch(?) iterations to run,
+# typically 20
+num_msmc_iterations = config["num_msmc_iterations"]
+
+num_sampled_genomes_per_replicate = config["num_sampled_genomes_per_replicate"]
+
+# The number of replicates of each analysis you would like to run
+# For now leaving it a 1 just to get results quickly
+replicates = config["replicates"]
+
+# Where you would like all output files from analysis to live
+output_dir = os.path.abspath(config["config"])
+
+# The analysis species
+species = stdpopsim.get_species(config["species"])
+
+# The specific model you would like to run
+if(config["model"] == "PiecewiseConstantSize"):
+    model = stdpopsim.PiecewiseConstantSize(species.population_size)
+    generation_time = species.generation_time
+else:
+    model = species.get_demographic_model(config["model"])
+    # For plotting
+    generation_time = model.generation_time
+
+# The genetic map you would like to use.
+# if value None is given default_recombination_rates are
+# used with a flat map
+genetic_map_id = config.get("genetic_map", None)
+
```

Comment:
then we can get the values for the annotation and the DFE from the config file here, just as with the genetic map
```suggestion
dfe_id = config.get("dfe_id", None)
annotation_id = config.get("annotation_id", None)
```

Reply:
done